wcivf/apps/elections/helpers.py

```diff
@@ -79,3 +81,16 @@ def __call__(self, request, *args, **kwargs):
 
         self.__name__ = self.__qualname__ = view.__name__
         return view(request, *args, **kwargs)
+
+
+def expected_sopn_publish_date(slug):
+    if not expected_sopn_publish_date.lookup:
+        expected_sopn_publish_date.lookup = StatementPublishDate()
+
+    try:
+        return expected_sopn_publish_date.lookup.for_id(slug)
+    except AmbiguousElectionId:
+        return None
```

Comment:
I accept this is probably currently non-trivial with the data we hold in WCIVF, but given the vast majority of elections we cover on WCIVF are local, the most common case is that this will throw `AmbiguousElectionId` and return `None`. For example, if we had this in place a week or two ago, this output would only be shown for something like 6 of the 5,000+ ballots happening on 2nd May.

In order to make this return a useful answer in most cases, we will need to find a way to derive the country.

Reply:
Agreed.

To my mind, the easiest thing would be keeping keeping a list of "these wards are Scotland/Northern Ireland" and assuming everything else is England and Wales, but that's probably not sustainable.

I suppose the future-proof way of doing this would be to add country to the WDIV API so it can be pulled into WCIVF that way? WDIV feels like the correct place for country to exist.

Reply:
We do store `territory_code` in EE, e.g https://elections.democracyclub.org.uk/api/elections/local.tower-hamlets.lansbury.by.2019-02-07/

I expect a quick way of doing this would be to do something like the existing [`import_metadata` on the election import](https://github.com/DemocracyClub/WhoCanIVoteFor/blob/master/wcivf/apps/elections/management/commands/import_elections.py#L28) but for [`import_posts`](https://github.com/DemocracyClub/WhoCanIVoteFor/blob/master/wcivf/apps/elections/management/commands/import_posts.py).

(This would require a DB change too)

Reply:
Oh, sweet! I noticed there used to be "group" attached to the post object but that's now blank?

```sql
wcivf=> select "group", count(*) from elections_post group by 1 order by 2;
      group       | count 
------------------+-------
 Northern Ireland |    36
 Wales            |    40
 Scotland         |    59
 England          |   533
                  | 12005
```

If this field could be backfilled/made current, this would be even easier to pull out and I don't think a need for a migration?

Reply:
Had a stab at this below. Interesting outputs:

```
wcivf=> select territory, count(*) from elections_post group by 1;
 territory | count 
-----------+-------
           |   410
 SCT       |   441
 ???       |     2
 GBN       |  1300
 WLS       |   902
 ENG       |  9520
 NIR       |    98
(7 rows)
```

The two `???` are from election ids starting `tmp_local`, but there are 410 posts without corresponding PostElections. What's the case for these, is this due to e.g. redrawing wards?